exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverStatsExporter.java

```diff
@@ -189,21 +205,41 @@ public static void createAndRegisterWithProjectIdAndMonitoredResource(
    * @param exportInterval the interval between pushing stats to StackDriver.
    * @param monitoredResource the Monitored Resource used by exporter.
    * @throws IllegalStateException if a Stackdriver exporter is already created.
+   * @deprecated in favor of {@link #createAndRegisterWithConfiguration(StackdriverConfiguration)}.
    */
+  @Deprecated
   public static void createAndRegisterWithMonitoredResource(
       Duration exportInterval, MonitoredResource monitoredResource) throws IOException {
     checkNotNull(exportInterval, "exportInterval");
     checkNotNull(monitoredResource, "monitoredResource");
-    createInternal(null, ServiceOptions.getDefaultProjectId(), exportInterval, monitoredResource);
+    createInternal(null, null, exportInterval, monitoredResource);
+  }
+
+  /**
+   * Creates a Stackdriver Stats exporter with a {@link StackdriverConfiguration}.
+   *
+   * @param configuration the {@code StackdriverConfiguration}.
+   */
+  public static void createAndRegisterWithConfiguration(StackdriverConfiguration configuration)
```

Comment:
Should we just call this method "createAndRegister", since it will be the only way to initialize the exporter?

Reply:
Unfortunately that name is used by a previous factory method, so currently we can only use this name.

Reply:
Couldn't we have two overloads, `createAndRegister(Duration)` and `createAndRegister(StackdriverStatsConfiguration)`?  I don't think anyone would confuse the two methods, because one would be deprecated.

Reply:
You're right, just forgot this is an overload method. Updated.